src/routes/Onboarding.svelte

```diff
@@ -0,0 +1,81 @@
+<script>
+  import { onMount, getContext, createEventDispatcher } from "svelte";
+  import Layout from "../components/Layout.svelte";
+  import Main from "../components/Main.svelte";
+  import Welcome from "../routes/welcome/Content.svelte";
+  import Demographics from "./demographics/Content.svelte";
+  import TermsOfService from "./terms-of-service/Content.svelte";
+  import OnboardingCTAContainer from "../components/OnboardingCTAContainer.svelte";
+
+  import DemographicsCallToAction from "./demographics/CallToAction.svelte";
+  import TermsCallToAction from "./terms-of-service/CallToAction.svelte";
+
+  export let view = "welcome";
+  let mounted = false;
+
+  const store = getContext("rally:store");
+
+  const dispatch = createEventDispatcher();
+
+  onMount(() => {
+    mounted = true;
+  });
+
+  function send(next) {
+    view = next;
+    window.scrollTo(0, 0);
+  }
+
+  function step(w) {
+    if (w === "welcome") return 1;
+    if (w === "terms") return 2;
+    if (w === "demographics") return 3;
+    return 1;
+  }
+
+  function finishOnboarding() {
+    dispatch("onboarding-complete");
+  }
+
+  let results;
+</script>
+
+{#if mounted}
+  <div>
+    <Layout>
+      <Main padForOnboarding={view !== 'welcome'}>
+        {#if view === 'welcome'}
+          <Welcome on:get-started={() => send('terms')} />
+        {:else if view === 'terms'}
+          <TermsOfService />
+        {:else if view === 'demographics'}
+          <Demographics bind:results />
+        {/if}
+      </Main>
+      <OnboardingCTAContainer
+        step={step(view)}
+        transparent={view === 'welcome'}>
+        {#if view === 'terms'}
+          <TermsCallToAction
+            on:accept={() => {
+              // Join Rally.
+              store.updateIonEnrollment(true);
+              send('demographics');
+            }}
+            on:cancel={() => {
+              send('welcome');
+            }} />
+        {:else if view === 'demographics'}
+          <DemographicsCallToAction
+            on:save={() => {
+              // Submit Demographics here.
+              // store.updateDemographicSurvey(results)
```

Comment:
@Dexterp37 Once we have a way to submit the demographic survey, this is where the store function would go.

Reply:
Ok, I can proceed as soon as I have `result` :)